src/shims/fs.rs

```diff
@@ -1118,6 +1115,13 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     }
 
     fn fsync(&mut self, fd_op: OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
+        // On macOS, fsync does not wait for the underlying disk to finish writing, while
+        // fcntl with the F_FULLFSYNC operation does. If fsync is somehow called by a macOS
+        // target, we will call File::sync_all(), which is a stronger operation that does
+        // wait for the disk to flush its buffers. (If the host is on Linux, the standard
+        // library will call fsync, as it does flush disk buffers, and if the host is also
+        // on macOS, it will call fcntl with F_FULLFSYNC)
```

Comment:
```suggestion
        // library will call fsync, as it does flush disk buffers, and if the host is
        // on macOS, it will call fcntl with F_FULLFSYNC.)
```
I am confused by the "also" here, how can it be Linux and "also" macOS?

Reply:
I had meant that in the sense that both the target and the host were macOS, I'll clarify it.